core/chains/evm/logpoller/orm.go

```diff
@@ -61,8 +61,9 @@ type ORM interface {
 	SelectLogsDataWordRange(ctx context.Context, address common.Address, eventSig common.Hash, wordIndex int, wordValueMin, wordValueMax common.Hash, confs evmtypes.Confirmations) ([]Log, error)
 	SelectLogsDataWordGreaterThan(ctx context.Context, address common.Address, eventSig common.Hash, wordIndex int, wordValueMin common.Hash, confs evmtypes.Confirmations) ([]Log, error)
 	SelectLogsDataWordBetween(ctx context.Context, address common.Address, eventSig common.Hash, wordIndexMin int, wordIndexMax int, wordValue common.Hash, confs evmtypes.Confirmations) ([]Log, error)
+
 	// FilteredLogs accepts chainlink-common filtering DSL.
-	FilteredLogs(filter query.KeyFilter, sortAndLimit query.LimitAndSort) ([]Log, error)
+	FilteredLogs(ctx context.Context, filter query.KeyFilter, limitAndSort query.LimitAndSort) ([]Log, error)
```

Comment:
There is `ObservedORM` decorator that keeps monitoring all LogPollers queries and publishes per query details (durations, number of records returned) to Prometheus. Having these metrics in Prometheus is critical for us. We need to keep a similar functionality for ChainReader/NewLogPoller, otherwise we will be blind to the queries performance

Reply:
If we want to keep the existing mechanism for publishing metrics probably something like this would be sufficient

```go
FilteredLogs(ctx context.Context, filter query.KeyFilter, limitAndSort query.LimitAndSort, queryName string) ([]Log, error)
```

and then
```go
func (o *ObservedORM) FilteredLogs(ctx context.Context, filter query.KeyFilter, limitAndSort query.LimitAndSort, queryName string) ([]Log, error) {
	return withObservedQueryAndResults(o, queryName, func() ([]Log, error) {
		return o.ORM.FilteredLogs(ctx, filter, limitAndSort, queryName)
	})
}
```

I believe having some sort of `name` or `alias` for the query would be very convenient for debugging as well (e.g. slow queries logger). `queryName` could be optional to not enforce naming when it's not needed, but I believe that we want to keep tracking all the DB queries

Reply:
> We have the contract/event name coming from the binding.  

Grouping per contract/event probably does make sense, but the same event/contract can be used in multiple LogPoller queries, so monitoring queries (SQL that is actually run on the DB) might be hard and not obvious. So far, most of the LP issues were either related to missing indexes or poorly written SQLs - IMO we want to keep track of these. I'm open to any solutions as long as we can track durations on the query level, maybe there are better ways than just adding `queryName` to the interface ;)

Reply:
This is now implemented in https://github.com/smartcontractkit/chainlink/pull/12793